src/software/ai/hl/stp/play/offense/offense_play.cpp

```diff
@@ -35,27 +35,31 @@ void OffensePlay::updateTactics(const PlayUpdate &play_update)
         num_defenders     = play_update.num_tactics - 1;
     }
 
-    shoot_or_pass_play->updateTactics(
-        PlayUpdate(play_update.world, num_shoot_or_pass,
-                   [&tactics_to_return](PriorityTacticVector new_tactics) {
-                       for (const auto &tactic_vector : new_tactics)
-                       {
-                           tactics_to_return.push_back(tactic_vector);
-                       }
-                   }));
+    shoot_or_pass_play->updateTactics(PlayUpdate(
+        play_update.world, num_shoot_or_pass,
+        [&tactics_to_return](PriorityTacticVector new_tactics) {
+            for (const auto &tactic_vector : new_tactics)
+            {
+                tactics_to_return.push_back(tactic_vector);
+            }
+        },
+        play_update.inter_play_communication,
+        play_update.set_inter_play_communication_fun));
 
     crease_defense_play->updateControlParams(
         play_update.world.ball().position(),
         TbotsProto::MaxAllowedSpeedMode::PHYSICAL_LIMIT);
 
-    crease_defense_play->updateTactics(
-        PlayUpdate(play_update.world, num_defenders,
-                   [&tactics_to_return](PriorityTacticVector new_tactics) {
-                       for (const auto &tactic_vector : new_tactics)
-                       {
-                           tactics_to_return.push_back(tactic_vector);
-                       }
-                   }));
+    crease_defense_play->updateTactics(PlayUpdate(
+        play_update.world, num_defenders,
+        [&tactics_to_return](PriorityTacticVector new_tactics) {
+            for (const auto &tactic_vector : new_tactics)
+            {
+                tactics_to_return.push_back(tactic_vector);
+            }
+        },
+        play_update.inter_play_communication,
+        play_update.set_inter_play_communication_fun));
```

Comment:
is it a potential problem that `crease_defense_play`'s callback overwrites the `inter_play_communication` set by `shoot_or_pass_play` ?

Reply:
yeah, it's currently not a problem since crease_defense_play doesn't invoke that callback, but it could be a problem

Reply:
i see. Feels a bit wrong though to force any modification of `crease_defense_play` to have to remember this

Reply:
no, offense play can manage this by passing in a different callback and arbitrate

Reply:
yeah i get that, i just mean if we dont have arbitration now then I hope we remember to add it when we modify  `crease_defense_play`